### PR TITLE
added webicity protocol

### PR DESCRIPTION
**Pull request by PiotrJagla**
Webicity protocol completed.
I think that i found a bug.
When we have url "webicity://welcome" the url.getPath() returns just "/" rather than "/welcome" so i extracted the path manually.
I think that this bug can appear due to lack of extension in path. If this is a case i can take care of it because it seems to be trivial problem.

### CODE REVIEW

File: src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	@Override
+	public String[] getSchemes() {
+		return new String[]{"webicity"};
+	}
+
+	@Override
+	public Connection openConnection(URL url, ProtocolContext context) throws IOException {
+
+		StringBuilder path = new StringBuilder(
+			url.toString().substring( url.getHost().length() + 3)
```
Comment (JasonTheKitten) [inline src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java]: It seems like the path actually gets treated like a host in this case. So you can just do url.getHost() for before the slash
If you would like any path after the slash, url.getHost() + "/" + url.getPath() should work.

I would not recommend converting the URL to a string and slicing it like this, as that is error prone

File: src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.github.webicitybrowser.webicity.core.net.Connection;
+import com.github.webicitybrowser.webicity.core.net.Protocol;
+import com.github.webicitybrowser.webicity.core.net.ProtocolContext;
+
+import java.io.*;
+
+public class WebicityProtocol implements Protocol {
+	@Override
+	public String[] getSchemes() {
+		return new String[]{"webicity"};
```
Comment (JasonTheKitten) [inline src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java]: Stylistic choice, but can you add spaces between `[]` and `{`, as well as between the brackets and string?

File: src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package com.github.webicitybrowser.webicitybrowser.engine.protocol;
+
+import com.github.webicitybrowser.spec.url.URL;
+import com.github.webicitybrowser.webicity.core.net.Connection;
+import com.github.webicitybrowser.webicity.core.net.Protocol;
+import com.github.webicitybrowser.webicity.core.net.ProtocolContext;
+
+import java.io.*;
```
Comment (JasonTheKitten) [inline src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java]: I am not using import wildcards in this project

**Review by JasonTheKitten — CHANGES_REQUESTED** [review]
Overall, looks great.
Just two stylistic nitpicks and one error-prone piece of code.

**Comment by PiotrJagla** [comment]
Thanks for review, i will fix this as i have time

File: src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java (hunk 1 above)
Comment (PiotrJagla) [reply src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java]: i saw that when there is only file name the "/" should be removed but when there is path like "src/index" doing as you said works fine. I committed changes and they are in this pull request i suppose.

**Pull request merged**